modal/_utils/deprecation.py

```diff
@@ -122,3 +124,40 @@ def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
         return func(*args, **kwargs)
 
     return wrapper
+
+
+# Utilities for deprecating the namespace parameter across Modal resources
+class _ArgumentNotPassedType:
+    """Sentinel object to detect when an argument is explicitly passed vs using default."""
+
+    def __repr__(self) -> str:
+        return f"{__name__}._ARGUMENT_NOT_PASSED"
+
+
+_ARGUMENT_NOT_PASSED = _ArgumentNotPassedType()
+
+
+def warn_if_passing_namespace(
+    namespace,
+    resource_name: str = "Modal resource",
+) -> "api_pb2.DeploymentNamespace.ValueType":
+    """Issue deprecation warning for namespace parameter and return appropriate default.
+
+    Args:
+        namespace: The namespace parameter value (may be sentinel or actual value)
+        resource_name: Name of the resource type for the warning message
+
+    Returns:
+        The appropriate namespace value to use
+    """
+    if namespace is _ARGUMENT_NOT_PASSED:
+        return api_pb2.DEPLOYMENT_NAMESPACE_WORKSPACE
```

Comment:
Do we need to actually be returning something here or should we just be warning when the user is passing something to `namespace=` in the Python interface? AFAIK for the object types you're touching here you can't actually accomplish anything productive by passing `namespace=`, so I don't think we actually need go out of our way to _use_ the value they passed.

The situation is indeed a little different for `modal.Image`, although even there I don't think there's a well-defined use for users passing some `namespace`.